### ISSUE TITLE
Allow changing the webview background color 

### ISSUE THREAD

**Issue by mallendeo** [tauri-apps/tauri#1564]
**Describe the bug**
Window flashes a white background before rendering app content on macOS Big Sur.

The css for the `index.html` is:

```css
html, body {
  width: 100%; height: 100%;
  background: rgb(231, 232, 232);
}

@media (prefers-color-scheme: dark) {
  html, body {
    background: rgb(37, 38, 40);
  }
}
```

**To Reproduce**
1. Create an index.html with `<style>html, body { background: black; }</style>`
2. Run the app (dev, or dist bundle)
3. Screen will flash a white background before rendering the content.

**Expected behavior**
The main window should wait for the content to load, or it should apply the page's background color on open.

**Screenshots**

https://user-images.githubusercontent.com/1578018/115579993-b0ccbf80-a294-11eb-8c68-07898105ab25.mp4

**Platform and Versions (please complete the following information):**

OS: Mac OS, version 11.2.3 X64
Node: 15.13.0
NPM: 7.7.6
Yarn: 1.22.10
Rustc: 1.51.0

**Additional context**
This is running the app with the default values (no transparent window mode). When using transparent window mode, it doesn't flash, but the window loses all shadows and borders until it is resized.

https://user-images.githubusercontent.com/1578018/115580068-bfb37200-a294-11eb-832d-9639839ae126.mp4

**Comment by kinghat** [comment]
also experiencing this on linux:

https://user-images.githubusercontent.com/16073360/115911342-94b35480-a433-11eb-90b9-23e64960c7a3.mp4

**Comment by chippers** [comment]
I suspected this is the browser's default background color, so I made a small edit to `wry` to set the background manually to red (linux only) to see this a bit easier.  The initial color (dark/light) is the OS window's default color, then the webview is loaded with whatever background color is set (red here), and then the application loads.

https://user-images.githubusercontent.com/3301043/115920054-844cab00-a42e-11eb-99e6-1ff5ddbd674f.mp4

This is a change needed in wry to support setting the webview background color.  I've opened up a ticket at https://github.com/tauri-apps/wry/issues/197.

If/when that is implemented, then we can look at expanding the `tauri::Attributes` trait to include setting the background color, and maybe add a hex string `backgroundColor` field to the window config inside `tauri.config.json`.

**Comment by happyshovels** [comment]
Since I also ran into this issue today, I am sharing my hacky workaround in case someone is also bothered by the white flashing. The idea is to start the window as hidden and then show it after the html is hopefully rendered.

It works somehow like this:

```
.setup(|app| {

      let main_window = app.get_window("main").unwrap();
      // we perform the initialization code on a new task so the app doesn't freeze
      tauri::async_runtime::spawn(async move {
        // adapt sleeping time to be long enough
        sleep(Duration::from_millis(500));
        main_window.show().unwrap();
      });

      Ok(())
    })
```
and win the tauri.conf.json
```
"windows": [
  {
    ...
    "visible": false
  }
]

**Comment by elibroftw** [comment]
> 

```rs
use std::thread::sleep;
use std::time::Duration;
```

**Comment by martpie** [comment]
I faced the same issue building Museeks with Electron. I am not sure there is much Tauri can do about it.

How I actually solved it is by hiding the window by default, and showing it only after my frontend app has loaded (via an event) (instead of using an arbitrary `sleep` that may need different values on different systems)

**Comment by isopterix** [comment]
@martpie: would you mind sharing your code?

**Comment by martpie** [comment]
Of course:

Tauri conf:

```diff
"windows": [
  {
    "title": "Your app",
+   "visible": false,
    "fullscreen": false,
    "resizable": true
  }
 ]
```

Rust side:

```rs
#[tauri::command]
pub async fn show_main_window(window: tauri::Window) {
    window.get_window("main").unwrap().show().unwrap(); // replace "main" by the name of your window
}
```

From your front-end, once you know your app is instantialized (for example with SolidJS):

```ts
import { invoke } from '@tauri-apps/api';

// Solid JS speicifc, `componentDidMount` for React, etc...
onMounted(() => {
  invoke('show_main_window');
})
```

you just have to find the equivalent of Vue, React etc if you use any other

**Comment by JonasKruckenberg** [comment]
> Of course:
> 
> 
> 
> Tauri conf:
> 
> 
> 
> ```diff
> 
> "windows": [
> 
>   {
> 
>     "title": "Your app",
> 
> +   "visible": false,
> 
>     "fullscreen": false,
> 
>     "resizable": true
> 
>   }
> 
>  ]
> 
> ```
> 
> 
> 
> Rust side:
> 
> 
> 
> ```rs
> 
> #[tauri::command]
> 
> pub async fn show_main_window(window: tauri::Window) {
> 
>     window.get_window("main").unwrap().show().unwrap(); // replace "main" by the name of your window
> 
> }
> 
> ```
> 
> 
> 
> From your front-end, once you know your app is instantialized (for example with SolidJS):
> 
> 
> 
> ```ts
> 
> import { invoke } from '@tauri-apps/api';
> 
> 
> 
> onMounted(() => {
> 
>   invoke('show_main_window');
> 
> })
> 
> ```
> 
> 
> 
> you just have to find the equivalent of Vue, React etc if you use any other

This is in fact the recommendations solution to this problem. And optimizing your Frontend rendering (through ssg or similar) so the delay to first-content is very small. Something like the app-shell architecture helps here as well.

**Comment by mantou132** [comment]
MacOS solution:

We can remove the webView background:

```rust
main_window.with_webview(|webview|  unsafe {
   let id = webview.inner();
   let no: cocoa::base::id = msg_send![class!(NSNumber), numberWithBool:0];
   let _: cocoa::base::id =
   msg_send![id, setValue:no forKey: NSString::alloc(nil).init_str("drawsBackground")];
});
```

ref: https://github.com/tauri-apps/wry/blob/765fe5ae413a1c13ad99802b49f3af859a2445d5/src/webview/macos/mod.rs#L180

then set the window background to the color you want:

```rust
let id = main_window.ns_window().unwrap() as cocoa::base::id;
let color = NSColor::colorWithSRGBRed_green_blue_alpha_(nil, 0.0, 0.0, 0.0, 1.0);
let _: cocoa::base::id = msg_send![id, setBackgroundColor: color];
```

**Comment by jlgerber** [comment]
Unfortunately, this doesn't seem to work for dynamically created windows. At least with in Sveltekit.  If you create a window using WebveiwWindow and set display: false in the options, then the onMount function never gets called for the widget with the supplied route.

**Comment by aiyanbo** [comment]
https://github.com/pretty-clean/bugs/issues/10

**Comment by Raduc4** [comment]
Has anyone solved this?

**Comment by reanimatedmanx** [comment]
Hey 👋 What if we extend the tauri config to support setting a background color from one single place to both the shell's background color and the webview's background color? So whenever the initial load happens, there would be no "flash granade" thrown into your face.

Then, if needed, it can be overriden by the webview's CSS styles.

So it would be:

1. Tauri's Config -> `appBackgroundColor` 
- Sets the background color for the app shell
- Sets the background color for the webview's browser default background color
2. Application specific css overrides 
- This would override the webview's browser defaults with whatever color you've picked

**Comment by KirillTregubov** [comment]
Just wanted to check in on what the status of this is in 2024. I am still experiencing the issue outlined by the author on Windows 11. The maintainers of wry have added [`set_background_color`](https://docs.rs/wry/latest/wry/struct.WebView.html#method.set_background_color) to `wry::Webview` in [version 0.21.0](https://github.com/tauri-apps/wry/releases/tag/wry-v0.21.0). I just checked my `Cargo.lock` file and the latest version of Tauri seems to be using `wry 0.24.7`.

Unfortunately, from my research, there seems to be no way for us to accesss this wry API in our Rust code. Is there any plan on exposing this functionality for Tauri users? Alternatively, has anyone found a solution to this issue?

**Comment by thnee** [comment]
It's an interesting problem, that is hard to solve. Even some major products still have the white flash on startup problem, like Google Chrome for example.

If we just define a color through a static config file, then it would still flash in the wrong color for users that prefer a different theme. This is less than ideal, but I guess that in general it's at least less annoying with black than white.

A somewhat better solution would be if Tauri could detect light/dark system theme, and set a configurable light/dark background color when creating the initial window. But if an app supports multiple themes, besides just light and dark, then it would still flash in the wrong color for users that prefer a different theme.

There is always going to be a delay between the initial window being created, and frontend code rendering colors. As long as the app is applying any customizable colors through frontend code, there will be a flash of a different color. So ideally there would be some more sophisticated solution here.

One option would be to add some kind of argument or hook to the window creation where a color can be provided as input. So that the app has a chance to read user and system settings and make a decision based on some custom logic, and pass that into the window creation.

It would be really cool if there was a whole theme system in Tauri, where we could define custom themes in config, and it would be integrated, so it applies automatically to window background and decorations, and is made available to the app code as variables in Rust, JS, and CSS.

---

For Tauri v2, I am using the following workaround. However, I belive this only works for desktop, and there is no equivalent thing for mobile, so the white flash still remains an issue there.

#### Tauri `tauri.conf.json`

```
{
	"app": {
		"windows": [
			{
				"title": "MyTitle",
				"visible": false
			}
		],
	},
```

#### SvelteKit `+layout.svelte`

```svelte
<script>
	import { onMount } from "svelte";

	import { WebviewWindow } from "@tauri-apps/api/webviewWindow";
	import { PhysicalSize } from "@tauri-apps/api/dpi";

	import globals from "$lib/globals.svelte";

	function setPreferedTheme() {
		let prefersDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
		let value = prefersDark ? "dark" : "light";
		document.documentElement.setAttribute("data-theme", value);
	}

	async function initWindow() {
		if (globals.windowInited) {
			return;
		}

		setPreferedTheme();

		// Sleep because it takes a split second for colors to update after setting theme,
		// which leads to a white flash if using dark theme.
		await new Promise(r => setTimeout(r, 42));

		let window = WebviewWindow.getCurrent();
		await window.setTitle("MyTitle");
		await window.setSize(new PhysicalSize(1400, 800));
		await window.setMinSize(new PhysicalSize(300, 400));
		await window.center();
		await window.show();

		globals.windowInited = true;
	}

	onMount(() => {
		initWindow();
	});
</script>
```

**Comment by pronebird** [comment]
It would be great to expose the background color setting in the Webview API. Spent about an hour trying to set background color here and there and even used some sample from with_webview which returns a platform webview but nothing really worked and the app still flashes white background at open.

**Comment by FabianLars** [comment]
> but nothing really worked and the app still flashes white background at open.

Likely because it's not (only) the webview flashing but also the native window itself which makes this quite a bit harder.

**Comment by pronebird** [comment]
> > but nothing really worked and the app still flashes white background at open.
> 
> 
> 
> Likely because it's not (only) the webview flashing but also the native window itself which makes this quite a bit harder.

Good point. I will set the background colour on both and see if it helps.

**Comment by pronebird** [comment]
I can create a window without webview and change the color. That works fine. 

But webview doesn't seem to react to the change of color:

```fs
pub trait WebviewExt {
    #[cfg(target_os = "macos")]
    fn set_background_color(&self, color: &tauri::window::Color) -> tauri::Result<()>;
}

impl WebviewExt for tauri::Webview {
    #[cfg(target_os = "macos")]
    fn set_background_color(&self, color: &tauri::window::Color) -> tauri::Result<()> {
        use crate::color_ext::ColorExt;
        let (r, g, b, a) = color.as_nscolor_tuple();

        self.with_webview(move |platform_webview| {
            unsafe {
                let nscolor: cocoa::base::id =
                    msg_send![class!(NSColor), colorWithDeviceRed:r green:g blue:b alpha:a];
                let id = platform_webview.ns_window();
                let () = msg_send![id, setBackgroundColor: nscolor];
            }
        })
    }
}
```

**Comment by pronebird** [comment]
TBH I think I found a much simpler way to handle this and I simply wait for `on_page_load()` to fire the `Finished` event, then I display the window. So far so good and then the window appears with CSS applied to webview etc..

**Comment by rogue-ninja-creative** [comment]
It seems apps using Electron have fixed this. I've been using MongoDB Compass for a long time and it had the same issue; a flash of white before the content loads and the background colour updates. But that seems to have been fixed with the latest  update.

**Comment by albingroen** [comment]
Any update on this? This is another one of those things that make me wanna go to Electron :/

I understand it's a hard problem, and if I knew where to look I'd love to help. Is anyone actively working on it?

The "white flash" effect in a dark mode app really makes it feel cheap...

**Comment by martpie** [comment]
The best workaround is to hide the window by default, and only show it once your app has loaded the required resources (and eventually painted another frame via requestAnimationFrame)

This may introduce an additional delay to show the app though.

Pick your poison.

**Comment by timuric** [comment]
There is a guide on how to do it at least on macos in tauri v2 https://v2.tauri.app/plugin/window-customization/#macos-transparent-titlebar-with-custom-window-background-color

**Comment by xuchaoqian** [comment]
> The "white flash" effect in a dark mode app really makes it feel cheap...

@albingroen I agree with your point, this is a very impactful issue for the user experience. Just imagine the feeling of opening VSCode every time and seeing a blank white screen.

@amrbashir has made efforts in the past: [feat: allow setting webview bg color](https://github.com/tauri-apps/wry/pull/682), and I feel we are quite close to solving this problem, we just need a hero to step up and conquer this challenge.

**Comment by martpie** [comment]
Do you mean it's now supported by Wry-vanilla and the only thing left is to implement this into Tauri's APIs? (Surface option in window configuration, pass it to Wry, etc)?

**Comment by its-monotype** [comment]
Is there anything similar to Electron's event `ready-to-show`? [ref](https://www.electronjs.org/docs/latest/api/browser-window#showing-the-window-gracefully)

**Comment by deminearchiver** [comment]
> Is there anything similar to Electron's event `ready-to-show`? [ref](https://www.electronjs.org/docs/latest/api/browser-window#showing-the-window-gracefully)

What I do is I make the window be initially invisible, attach either a `DOMContentLoaded` event listener on the JS side, or use a framework-specific "on mount" hook, like `createEffect` for React or `onMount` for SolidJS, in which show the current window.

**Comment by KirillTregubov** [comment]
With the release of tauri v2, is there any update to this?

I'll try digging through the changes and see if I can find something to accomplish this.

**Comment by dionysuzx** [comment]
I solved this issue in my Svelte 5 + Tauri 2.0 app like so:

1. Set the `"core:window:allow-show"` permission under `capabilities/default.json` like so:
    ```
    "permissions": ["core:default", "core:window:allow-show", "shell:allow-open"]
    ```

2. Show window on mount (using state rune / store):

    Homepage:
    ```
    <script lang="ts">
      import { onMount } from "svelte";
      import { window } from "../stores/window.svelte.js";
    
      onMount(() => {
        window.show();
      });
    </script>
    ```

    Writable store:
    ```
    import { getCurrentWebviewWindow } from "@tauri-apps/api/webviewWindow";
    
    let shown = $state(false);
    
    export const window = {
      get shown() {
        return shown;
      },
      async show() {
        if (!shown) {
          console.log("showing window");
          await getCurrentWebviewWindow().show();
          shown = true;
        } else {
          console.log("window already shown");
        }
      },
    };
    ```

Hope it helps, cheers! 🍺

**Comment by dionysuzx** [comment]
**EDIT: THIS DOESN'T WORK**

I solved this issue in my Svelte 5 + Tauri 2.0 app like so:

1. Set `"visible": false` in `tauri.conf.json`.

2. Set the `"core:window:allow-show"` permission under `capabilities/default.json` like so:
    ```json
    {
        ...
        "permissions": ["core:default", "core:window:allow-show", "shell:allow-open"]
    }
    ```

3. Show window on mount (using state rune / store):

    Homepage:
    ```svelte
    <script lang="ts">
      import { onMount } from "svelte";
      import { window } from "../stores/window.svelte.js";
    
      onMount(() => {
        window.show();
      });
    </script>
    ```

    Writable store:
    ```javascript
    import { getCurrentWebviewWindow } from "@tauri-apps/api/webviewWindow";
    
    let shown = $state(false);
    
    export const window = {
      get shown() {
        return shown;
      },
      async show() {
        if (!shown) {
          console.log("showing window");
          await getCurrentWebviewWindow().show();
          shown = true;
        } else {
          console.log("window already shown");
        }
      },
    };
    ```

Hope it helps, cheers! 🍺

**Comment by xuchaoqian** [comment]
You will see the white background again if you reload, refresh, or resize the window. It seems very abrupt, especially in dark theme mode.

> I solved this issue in my Svelte 5 + Tauri 2.0 app like so:
> 
> 1. Set `"visible": false` in `tauri.conf.json`.
> 2. Set the `"core:window:allow-show"` permission under `capabilities/default.json` like so:
>    ```json
>    {
>        ...
>        "permissions": ["core:default", "core:window:allow-show", "shell:allow-open"]
>    }
>    ```
> 3. Show window on mount (using state rune / store):
>    Homepage:
>    ```svelte
>    <script lang="ts">
>      import { onMount } from "svelte";
>      import { window } from "../stores/window.svelte.js";
>    
>      onMount(() => {
>        window.show();
>      });
>    </script>
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    Writable store:
>    ```js
>    import { getCurrentWebviewWindow } from "@tauri-apps/api/webviewWindow";
>    
>    let shown = $state(false);
>    
>    export const window = {
>      get shown() {
>        return shown;
>      },
>      async show() {
>        if (!shown) {
>          console.log("showing window");
>          await getCurrentWebviewWindow().show();
>          shown = true;
>        } else {
>          console.log("window already shown");
>        }
>      },
>    };
>    ```
> 
> Hope it helps, cheers! 🍺

**Comment by dionysuzx** [comment]
you're right @xuchaoqian. this is not fixed. i'll update my comment until i figure out a fix... unless; do you know how to fix it? :)

**Comment by xuchaoqian** [comment]
We need [this PR](https://github.com/tauri-apps/wry/pull/682)  to be applied to Tauri.

> you're right @xuchaoqian. this is not fixed. i'll update my comment until i figure out a fix... unless; do you know how to fix it? :)

**Comment by meow2149** [comment]
You just do it:

```
import { getCurrentWebviewWindow } from '@tauri-apps/api/webviewWindow'
window.addEventListener('DOMContentLoaded', async () => {
  await getCurrentWebviewWindow().show()
})
```

**Comment by ScepticDope** [comment]
Not working on Windows, non of the workarounds work either.

Also on resizing the window you can see the white background flashing, you'd think at least that would be fixed by adding a backgroundColor?

Mentioned here: https://github.com/tauri-apps/tauri/issues/5170#issuecomment-2559585065

Can this issue please be opened again?

**Comment by rogue-ninja-creative** [comment]
> Not working on Windows, non of the workarounds work either.
> 
> Also on resizing the window you can see the white background flashing, you'd think at least that would be fixed by adding a backgroundColor?
> 
> Mentioned here: [#5170 (comment)](https://github.com/tauri-apps/tauri/issues/5170#issuecomment-2559585065)
> 
> Can this issue please be opened again?

I thought this was fixed in version 2.1

**Comment by ScepticDope** [comment]
@rogue-ninja-creative No its not fixed sadly. Just updated to 2.1.1 and checked, there is even new buggy behaviour now, the window menu also really slowly pops in now before the rest is rendered afterwards. Here is a basic working hack of my own for Vanilla Javascript. This won't fix the white flashing while resizing, only on startup. If your software has multiple windows just make it accept a window label param:

### tauri.conf.json
```
...
"windows": [
      {
        "label": "YourWindowLabelName",
        "visible": false,
        ...
```
_*`YourWindowLabelName`, the default is `main`._

### lib.rs
```
// imports
...
// Ugly HACK! Remove when possible see:
// https://github.com/tauri-apps/tauri/issues/1564
// https://github.com/tauri-apps/tauri/issues/5170
#[tauri::command]
fn show_window(app: tauri::AppHandle) {
    app.get_webview_window("YourWindowLabelName").unwrap().show().unwrap();
}
...
tauri::Builder::default()
        .setup(|app| {
        ...
        })
        ...
        .invoke_handler(tauri::generate_handler![show_window])
```

### main.js
```
...
const { invoke } = window.__TAURI__.core;
...
window.addEventListener("DOMContentLoaded", () => {
  invoke("show_window");
  ...
```

**Comment by petersamokhin** [comment]
This issue should be still `Open` or at least closed as `Not planned` because it doesn't work for the major platforms (macOS at least).

If anyone discovers this issue, I made it work for me on macOS:

1. Use private macOS APIs and "transparent" when creating your window.
2. Set `:root, html, body` background color in CSS
3. Set the window background color using the raw APIs from Tauri docs: https://tauri.app/learn/window-customization/#macos-transparent-titlebar-with-custom-window-background-color

=> no white blinking, no theme-independent grey blinking before page is loaded.

I tried to access `WKWebView` and use [`setUnderPageBackgroundColor`](https://developer.apple.com/documentation/webkit/wkwebview/underpagebackgroundcolor?language=objc) but it had no effect ([as per advice](https://github.com/tauri-apps/tauri/pull/11486#issuecomment-2483560242)) despite throwing no errors.

**Comment by alterhuman** [comment]
@petersamokhin new to tauri, can you tell me how to access the WKWebView? I want to try a solution I have in mind.

**Comment by Daydreamer-riri** [comment]
Hi everyone, if the above method still doesn't work for you, please check if you are using `tauri-plugin-window-state`. It remembers the `visible` property, causing your `visible(false)` setting to be ineffective.

You need to configure `tauri-plugin-window-state` like this:

```rust
use tauri_plugin_window_state::StateFlags;
tauri::Builder::default()
		.plugin(tauri_plugin_window_state::Builder::default().with_state_flags(StateFlags::all() & !StateFlags::VISIBLE).build())
        .run(tauri::generate_context!())
        .expect("error while running tauri application")
```

This allows `window-state` to ignore the `visible` property, making your `visible(false)` effective.

**Comment by cristianvogel** [comment]
I found this working for macOs and SvelteKit
In `app.html` 

set 

```
<style>
  html {
   background-color: #222;
  }
</style>
```

and 

```
 <div style="display: contents; background-color: #222 !important;">%sveltekit.body%</div>
```

and set the tauri window builder bg color dark as well on that back end

**Comment by Pandananana** [comment]
The same issue happens on Android. Do the solutions above also work for that?

**Comment by abose** [comment]
Hello, I would like to add that this is a breaker for light sensitive users who reported this issue as straining. It would be genuinely cool to address as this is a core accessibility issue that excludes some users from using the app. 

Since this is a core accessibility issue, request to reopen this issue even if a fix is not planned, as the issue remains open in Tauri 2.0 as well.

Please see tracking issue https://github.com/phcode-dev/phoenix/issues/2149 for user comments.

**Comment by dionysuzx** [comment]
> Hello, I would like to add that this is a breaker for light sensitive users who reported this issue as straining. It would be genuinely cool to address as this is a core accessibility issue that excludes some users from using the app.
> 
> Since this is a core accessibility issue, request to reopen this issue even if a fix is not planned, as the issue remains open in Tauri 2.0 as well.
> 
> Please see tracking issue [phcode-dev/phoenix#2149](https://github.com/phcode-dev/phoenix/issues/2149) for user comments.

Not to be annoying, but just +1-ing this. The white flicker is still an accessibility issue... especially with apps that have the mode dynamically selected based on the OS selection (eg. some are light mode, some are dark mode), so simply setting a background color does not seem to fix the issue.

**Comment by deminearchiver** [comment]
I have never bothered about not being able to change Webview background color for accessibility, because I always set the window to be hidden initially, and it is shown only when DOM content gets loaded (via something like framework effects or a plain `DOMContentLoaded` event.

Even in a multi-webview window it would be possible to defer showing until all webviews have finished loading.

**Comment by FabianLars** [comment]
^ No matter how well this flash can be fixed (don't get your hopes up btw), spawning the window hidden almost always makes sense. The first render(s) even without the white flash are always ugly with todays js frameworks etc. 🤷 

> Hello, I would like to add that this is a breaker for light sensitive users who reported this issue as straining. It would be genuinely cool to address as this is a core accessibility issue that excludes some users from using the app.

you should consider sueing Microsoft considering that Windows' windowing system had those white flashes ever since Windows 7 (with Windows Aero) 🙃 

---

If you want this to be 100% fixed we can only try to do the window/webview hiding for you at the moment (which tbh would be too much work to get right).

**I will not take another look at this issue until https://github.com/MicrosoftEdge/WebView2Feedback/issues/4953 is fixed.** (Edit: for windows)

---

Edit: Seeing that lately there were messages about other platforms as well:

**Android**: Setting the window's backgroundColor to the same color as your html goes a long way. The white (or dark, depending on dark mode settings) flash at the start that's left is basically android's built-in splashscreen.
For small-ish apps adding this to both themes.xml files in `gen/android/app/src/main/res/values(-night)` this should be enough (it was for me at least)
```xml
<item name="android:windowDisablePreview">true</item>
<item name="android:windowBackground">@null</item>
<item name="android:windowIsTranslucent">true</item>
```
^Not recommended by Google (did not find a source for that) but i've not seen issues with it in (small) tauri apps. Still, you may be more on the safe side removing the `windowDisablePreview` key and setting `windowBackground` to your app's color as well.

If your app gets bigger and thus needs more time to load into memory you'll probably have to look into the native splashscreen apis. It looks fairly complex though so i think we'll have to consider this out of scope for tauri.

**macOS**: Never really looked into it much but i'd recommend to use the privateMacOSApi config mentioned above to make the webview transparent. I heard that non-tauri apps still got approved to the macos store with that.

**iOS**: can't test this at the moment. did anyone else test it already?

**Comment by abose** [comment]
Thanks for looking into it @FabianLars . Looks like this is deadlocked as apps want a tauri fix and tauri is waiting for a fix on an issue from os vendors that may never come.

**Comment by FabianLars** [comment]
Okay i see, the first part of my message (the windows part) was written poorly. Should have taken more time for it.

The Windows 7 issue can be somewhat easily worked around if you know what you're doing in win32 land (i don't) but it doesn't make sense to look into it as long as the chromium "bug" is present in webview2. If you disagree then we're always happy to see contributions.

The chromium issue seems to be actively being worked on (somewhat fixed even, just not merged/shipped in edge) so i hope we'll see it in webview2 soon.

**Comment by dionysuzx** [comment]
> iOS: can't test this at the moment. did anyone else test it already?

@FabianLars I can confirm the white flicker happens on iOS, but is that what you need tested or something else? I'm happy to test anything with my iPhone that would help with this issue.

Thanks for taking a look.

**Comment by FabianLars** [comment]
Yes that's what i wanted you to test, thanks. I got hands on my macbook quicker than expected so i tested a bit as well.

There's a small-ish issue with the LaunchScreen rn, which for now seems to be a good thing. If that's "fixed" (which is something that has to be done in the xcode project either manually or by regenerating it after the next cli release) you'll have to configure the LaunchScreen color in xcode similar to the android story.

The real problem here though is a white/black flash between the launch screen and the webview load that i don't think i can fix (biggest iOS noob). Seems like capacitor faces this as well without a fix or workaround so that really makes me not hopeful https://github.com/ionic-team/capacitor/issues/3589
I'll still try to look into it once a while but i wouldn't count on me to fix it if i were you :)

The first issue helps to hide the second one as they both show up in the same color (white or black) for me so it just looks like a one second load screen instead of something flashing.

**Comment by dionysuzx** [comment]
> Yes that's what i wanted you to test, thanks. I got hands on my macbook quicker than expected so i tested a bit as well.
> 
> There's a small-ish issue with the LaunchScreen rn, which for now seems to be a good thing. If that's "fixed" (which is something that has to be done in the xcode project either manually or by regenerating it after the next cli release) you'll have to configure the LaunchScreen color in xcode similar to the android story.
> 
> The real problem here though is a white/black flash between the launch screen and the webview load that i don't think i can fix (biggest iOS noob). Seems like capacitor faces this as well without a fix or workaround so that really makes me not hopeful [ionic-team/capacitor#3589](https://github.com/ionic-team/capacitor/issues/3589) I'll still try to look into it once a while but i wouldn't count on me to fix it if i were you :)
> 
> The first issue helps to hide the second one as they both show up in the same color (white or black) for me so it just looks like a one second load screen instead of something flashing.

If the first issue is fixed what happens exactly? Are you saying it decreases the load time or something else happens? I don't fully understand what the first issue is.

And for the second issue, do you know if the white / black flash would be configurable? It seems the flash is always white for me.

And lastly if the first issue is fixed, are you saying the load screen is longer so it doesn't look like a flicker--if so, isn't it worse to have a longer load? I think I don't fully understand what you're saying about it though.

Thanks for looking into it. I do think that by being able to at least control the color of that flash or background color would be good to hide the issue more. Even better if it can be set from the theme selected in the OS.

**Comment by FabianLars** [comment]
> If the first issue is fixed what happens exactly? Are you saying it decreases the load time or something else happens?

timing wise nothing changes. currently i see a somewhat long white or black screen at the start (more on that later). If this issue is fixed, and let's say i changed the LaunchScreen to green then i first get a green launch screen, then a black or white flash, and then the app shows up.
The issue here is just that the launchscreen wasn't registered properly and therefore replaced by some default white/black screen 🤷 

If your project is old-ish then this may not even matter to you idk when it broke.

> And for the second issue, do you know if the white / black flash would be configurable? It seems the flash is always white for me.

well, that's what this issue here is all about really. I can only configure it between white and black via i _think_ it was the transparency setting. i doubt this is reliable across devices or settings though.

i can try to record it some time but i don't think this is really relevant and just a waste of time since it's not really about the actual flicker issue.

**Comment by PolarisSdesu** [comment]
I found a way to fix this at Tauri 2.0!!!
https://x.com/PolarisSdesu/status/1895442258478194831

**Comment by FabianLars** [comment]
Yes, the visibility workaround was mentioned a few times here and even without the flicker issue we would still recommend it to hide the first incomplete renders :)

**Comment by plitarko** [comment]
> Yes, the visibility workaround was mentioned a few times here and even without the flicker issue we would still recommend it to hide the first incomplete renders :)

Seems to only work for desktop, the Window.show() or WebviewWindow.show() methods both don't exist when targeting iOS.

**Comment by abose** [comment]
The issue with the hide show logic is that it will create a heavy weight perception of app at app start. In our measurements, windows first show delayed by half a second, which is about 20x higher than the 20ms start time we track currently. That's why we couldn't use that fix.

**Comment by PolarisSdesu** [comment]
> Yes, the visibility workaround was mentioned a few times here and even without the flicker issue we would still recommend it to hide the first incomplete renders :)

no way... i got an issue
I use this method, when you set windowEffect to acrylic, it becomes mica after the window is showed...
insane...

**Comment by FabianLars** [comment]
Then please open another GitHub issue for this... Ideally with reproduction steps

**Comment by dionysuzx** [comment]
> > Yes, the visibility workaround was mentioned a few times here and even without the flicker issue we would still recommend it to hide the first incomplete renders :)
> 
> Seems to only work for desktop, the Window.show() or WebviewWindow.show() methods both don't exist when targeting iOS.

Are you sure it only works on desktop? Can anyone confirm if there is no fix currently for mobile? cc @PolarisSdesu

**Comment by plitarko** [comment]
It does not compile for iOS with the error that the .show() method required for this work around doesn't exist

**Comment by FabianLars** [comment]
okay so the iOS flash is actually not so hard to fix (in handsight at least...) i'm just not sure how to integrate it well into tauri/wry.

https://github.com/user-attachments/assets/b2f1d3c7-a6ed-46c1-a11d-cd5789c832d8

Video explanation: green is LaunchScreen, orange is the black/white flash we're trying to bridge (and the top and bottom bars) and blue is the html/css color.

My first idea would be to simply re-use the backgroundColor webview/window config for the orange part but idk if that's weird with the bars at the top and bottom? I'm not a iOS user or developer so i'd appreciate some input if that's possible with the very limited info i provided.

**Comment by plitarko** [comment]
> My first idea would be to simply re-use the backgroundColor webview/window config for the orange part but idk if that's weird with the bars at the top and bottom? I'm not a iOS user or developer so i'd appreciate some input if that's possible with the very limited info i provided.

I think this solution looks perfectly fine! The blue rectangle in your screen is the HTML window and it can be set to go outside of its safe areas at the top and bottom. Additionally, I guess you would always want its color to match the orange one to get a smooth transition.

Let's gooo!! Great job!

**Comment by Innominus** [comment]
> I think this solution looks perfectly fine! The blue rectangle in your screen is the HTML window and it can be set to go outside of its safe areas at the top and bottom. Additionally, I guess you would always want its color to match the orange one to get a smooth transition.


@plitarko do you have a link to documentation about extending the HTML window/webview to go outside of the safe areas, and also a way to detect the safe area insets so we can say, set a background image from top to bottom but still have our inner UI sit safely within the safe area?

**Comment by Innominus** [comment]
> okay so the iOS flash is actually not so hard to fix (in handsight at least...) i'm just not sure how to integrate it well into tauri/wry.
> 
>  Simulator.Screen.Recording.-.iPhone.16.-.2025-03-05.at.23.22.55.mp4 
> Video explanation: green is LaunchScreen, orange is the black/white flash we're trying to bridge (and the top and bottom bars) and blue is the html/css color.
> 
> My first idea would be to simply re-use the backgroundColor webview/window config for the orange part but idk if that's weird with the bars at the top and bottom? I'm not a iOS user or developer so i'd appreciate some input if that's possible with the very limited info i provided.

@FabianLars this looks great! Will there also be a solution for Android with setting status bar/notch colours/webview background colour?

**Comment by plitarko** [comment]
@Innominus 

To make the html window ignore the safe areas, you need to add

`content="viewport-fit=cover"`

to your [viewport meta tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag)

Then you want to set the background image on something, e.g. your body element.

Whichever element has the background image needs to be a parent to the rest of your app, and you can now use something like 

` body {
 padding-top: env(safe-area-inset-top);
  padding-bottom: env(safe-area-inset-bottom);
  padding-left: env(safe-area-inset-left);
  padding-right: env(safe-area-inset-right);
}`

to make the rest of the app never bleed into the safe zones. You can also add overflow: hidden; to be extra sure.
Here is the documentation for the safe-area css: https://developer.mozilla.org/en-US/docs/Web/CSS/env

Hope that helps.

**Comment by voxelbee** [comment]
I was wondering if we could solve this on mobile or even desktop where we can have a native view shown over the top of the webview until an event is emitted to the native view saying that we have loaded? This would also allow us to animate or have a splash screen view that looks the same as the LaunchScreen for iOS. I'm sure android would be similar. This would fix anything to do with color switching and would be up to customizing the native view to how you want on each platform with icons too if you would like.

> My first idea would be to simply re-use the backgroundColor webview/window config for the orange part but idk if that's weird with the bars at the top and bottom? I'm not a iOS user or developer so i'd appreciate some input if that's possible with the very limited info i provided.

@FabianLars is this something I can try now on iOS?

**Comment by Innominus** [comment]
@plitarko 
Thanks! I found this information earlier today and it's really good on iOS. It looks like we'll be getting a solution to the colour flashing soon too, so that'll be great.
I am still trying to work out how to change the colour of the status icons on iOS though from black to white, and also how I can style android status bar/notch as well. If you have any ideas on that I'd be keen to hear :)

**Comment by plitarko** [comment]
@Innominus 

>  I am still trying to work out how to change the colour of the status icons on iOS though from black to white, and also how I can style android status bar/notch as well. If you have any ideas on that I'd be keen to hear :)

Pretty sure it does that automatically based on which one has the highest contrast, it's not something you can control.

**Comment by FabianLars** [comment]
> I was wondering if we could solve this on mobile or even desktop where we can have a native view shown over the top of the webview until an event is emitted to the native view saying that we have loaded? This would also allow us to animate or have a splash screen view that looks the same as the LaunchScreen for iOS. I'm sure android would be similar. This would fix anything to do with color switching and would be up to customizing the native view to how you want on each platform with icons too if you would like.

sounds like something i'd consider out of scope for now (cause too complicated).

> FabianLars is this something I can try now on iOS?

No, you'll have to wait at least until i push a branch to Wry.

**Comment by bradleat** [comment]
@FabianLars does this iOS solution use the visible trick? Will you link the branch here when you have it?

**Comment by bradleat** [comment]
Oh I see it:

https://github.com/tauri-apps/wry/pull/1513

**Comment by FabianLars** [comment]
> does this iOS solution use the visible trick?

not the hide/show stuff we talked about here if that's what you mean. We just set the transparency and colors correctly and as early as possible 🤷 

> Will you link the branch here when you have it?

no promises, i may forget.

> Oh I see it:
> https://github.com/tauri-apps/wry/pull/1513

i'd like to highlight the pr description because i already know that some people won't read it:
> in a typical tauri app there's still a one frame flicker left from tao due to windows' insane drawing behavior. will see if i can dig out my research on that, i think i had a fix but ditched it for Amr's drawing PR.

**Comment by Innominus** [comment]
> @Innominus 
> 
> >  I am still trying to work out how to change the colour of the status icons on iOS though from black to white, and also how I can style android status bar/notch as well. If you have any ideas on that I'd be keen to hear :)
> 
> Pretty sure it does that automatically based on which one has the highest contrast, it's not something you can control.

@FabianLars 
What about on Android? It seems iOS is solved with the viewport fit cover tag and your future background colour changes through the config json file, but on Android will this change the background colour behind the status bar and notch, and are there plans/a plugin to change the status/notch colours? Thanks Fabian :)

Edit: I realised just now you didn't write this comment, but I think the question is still relevant

**Comment by Innominus** [comment]
Just adding to my above specifically for Android, after some further investigation and tinkering you can actually set quite a lot of properties statically with the themes.xml inside of `gen/android/app/src/main/res/values/themes.xml.` Things such as `android:statusBarColor,` `android:navigationBarColor`, `android:windowBackground`, `android:background`, `android:windowLightStatusBar`, etc.
To stop from your splash screen colour to white to your background colour you can set the window transparent property to true and that seems to remove the white background of the container and allow the window background colour to come through.

**Comment by hydra1983** [comment]
@FabianLars I upgraded wry to 0.50.4, should I do something to make it working with iOS?
```
cargo tree|grep wry                                                                                                                                                                                             ─╯
│   ├── tauri-runtime-wry v2.4.1
│   │   └── wry v0.50.4
```

```
tauri info                                                                                                                                                                                                      ─╯

[✔] Environment
    - OS: Mac OS 15.2.0 arm64 (X64)
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.85.0 (4d91de4e4 2025-02-17)
    ✔ cargo: 1.85.0 (d73d2caf9 2024-12-31)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 20.14.0
    - pnpm: 10.4.1
    - npm: 10.7.0

[-] Packages
    - tauri 🦀: 2.3.1
    - tauri-build 🦀: 2.0.6
    - wry 🦀: 0.50.4
    - tao 🦀: 0.32.8
    - @tauri-apps/api : 2.2.0 (outdated, latest: 2.3.0)
    - @tauri-apps/cli : 2.2.7 (outdated, latest: 2.3.1)

[-] Plugins
    - tauri-plugin-log 🦀: 2.3.1
    - @tauri-apps/plugin-log : 2.2.2 (outdated, latest: 2.3.1)
    - tauri-plugin-opener 🦀: 2.2.6
    - @tauri-apps/plugin-opener : 2.2.5 (outdated, latest: 2.2.6)
    - tauri-plugin-store 🦀: 2.2.0
    - @tauri-apps/plugin-store : 2.2.0
    - tauri-plugin-notification 🦀: 2.2.2
    - @tauri-apps/plugin-notification : 2.2.1 (outdated, latest: 2.2.2)
    - tauri-plugin-http 🦀: 2.4.0
    - @tauri-apps/plugin-http : 2.3.0 (outdated, latest: 2.4.0)
    - tauri-plugin-fs 🦀: 2.2.0
    - @tauri-apps/plugin-fs : not installed!

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://192.168.0.102:1420/
    - framework: React
    - bundler: Vite
```

**Comment by FabianLars** [comment]
Please read my messages above. I did not push anything to fix iOS yet and it'll probablyyyy take a while until I do.

**Comment by hydra1983** [comment]
@FabianLars OK, got it.

**Comment by plitarko** [comment]
> Please read my messages above. I did not push anything to fix iOS yet and it'll probablyyyy take a while until I do.

Thank you for you great work!!
Is there any timeline for this? I have a Tauri app in Test Flight, but I don't want to make it public in the app store until this issue has been fixed.

**Comment by FabianLars** [comment]
My "while" is typically a long-ish timeframe or in other words: if i won't do it tomorrow it'll take at least a week.

**Comment by plitarko** [comment]
Following up to hear if this has been added yet, as it's been a couple of weeks now! Excited!!!

**Comment by bradleat** [comment]
@FabianLars also def looking for this so I can land the next iOS release of ToneTempo with a smooth startup experience. Is there anything I can do to speed this along? Would manually building a version wry from source from here (https://github.com/tauri-apps/wry/pull/1513) do the trick?

**Comment by FabianLars** [comment]
Still didn't push the iOS changes and can't give you an eta. A custom wry build would require you/someone to make the necessary changes first.

P.s. the pr you linked is for windows and already released.

**Comment by nathancovey** [comment]
Is a fix for this issue on macOS planned?

**Comment by ghufby** [comment]
I'm also having trouble changing the window background color in macOS.

What I did:

1. Started a new test project.

2. Removed all background color references from CSS.

3. Added backgroundColor property in tauri.conf.json.

```
"windows": [
    {
	"title": "Test",
	"width": 800,
	"height": 600,
        "backgroundColor": "#FF0000"
    }
],
```

4. Added setBackgroundColor permissions in tauri.conf.json.

```
"permissions": [
    "core:window:allow-set-background-color",
    "core:webview:allow-set-webview-background-color"
]

```
5. Added setBackgroundColor in main.js.
```

const { getCurrentWindow } = window.__TAURI__.window;
const { getCurrentWebview } = window.__TAURI__.webview;
const { getCurrentWebviewWindow} = window.__TAURI__.webviewWindow;

window.addEventListener("DOMContentLoaded", async () => {
    await getCurrentWindow().setBackgroundColor("#FF0000");
    await getCurrentWebview().setBackgroundColor("#FF0000");
    await getCurrentWebviewWindow().setBackgroundColor("#FF0000");
});
```

The window color is still white.

**Comment by 58bits** [comment]
> It seems apps using Electron have fixed this. I've been using MongoDB Compass for a long time and it had the same issue; a flash of white before the content loads and the background colour updates. But that seems to have been fixed with the latest update.

I still see a white flash in MongoDB Compass - even with the latest version on Windows 11 in dark mode.

**Comment by hamzaq2000** [comment]
In Tauri 2.0, the following works:

`tauri.conf.json`
```json
"windows": [
  {
    "title": "tauri-app",
    "width": 800,
    "height": 600,
    "visible": false
  }
]
```

`lib.rs`
```rust
#[tauri::command]
fn show_window(window: tauri::Window) -> Result<(), String> {
    window
        .show()
        .map_err(|e| format!("Failed to show window: {}", e))?;
    window
        .set_focus()
        .map_err(|e| format!("Failed to set focus: {}", e))?;
    Ok(())
}
```

`+page.svelte`
```javascript
import { onMount } from "svelte";

onMount(() => {
    setTimeout(async () => { await invoke("show_window") }, 50);
});
```

**Comment by FabianLars** [comment]
https://github.com/tauri-apps/wry/pull/1555 To test it, add this to your Cargo.toml
```toml
[patch.crates-io]
wry = { git = "https://github.com/tauri-apps/wry", branch = "feat/ios-bgcolor" }
```
Afterwards run `cargo update` and if it doesn't complain about something like "Patch for wry not applied because not in tree) it should work without changes (assuming you already set a background color via the window configs).

Note that you also have to change the LaunchScreen color to get rid of all flashes:

0) If your project was generated before tauri cli 2.3.1 you'll have to re-run `tauri ios init` or manually set the LaunchScreen's initialViewController (please ask google. i can't remember how that was done)
1) Open xcode in your project for example via `tauri ios dev --open` (opening it manually via xcode's project selector should work as well)
2) In the left project overview sidebar select yourprojectname > assets > LaunchScreen
3) In the middle window, select the `View` element.
4) While that element is selected there should be another sidebar on the right (if not, click the sidebar button in the upper right corner of the xcode window titlebar). There is a `background` option that you should set to the same color as your window color (or not lol)

**Comment by e-simpson** [comment]
@FabianLars Thanks for that! I tried that out on iOS and was able to colour the initial splash screen! How did you get to the level of control that is shown in https://github.com/tauri-apps/tauri/issues/1564#issuecomment-2702228866?

Currently on app load, I see the launch screen colour, then black, then white, then my app/browser/background colour. I've tried most everything mentioned in this thread to no avail.

**Comment by FabianLars** [comment]
>How did you get to the level of control that is shown in https://github.com/tauri-apps/tauri/issues/1564#issuecomment-2702228866?

With the branch i showed in the comment right above yours. That's handling the color between the launchscreen and your frontend color etc.

**Comment by e-simpson** [comment]
@FabianLars hmm, interesting. I applied that wry branch, updated cargo, re-built iOS but still see the black. Maybe I'm not setting the background correctly in my app.

**Comment by FabianLars** [comment]
Could also be the timing of that branch, i was a bit surprised that what i did worked (in the video above i had it slightly different).

To me it could be 3 things:
1) The initialViewController isn't set correctly. I don't think this can actually be the case if your app starts with the LaunchScreen.
2) The timing of my branch is broken, or the branch isn't applied in your project (cargo should tell you about it)
3) The html background is wrong or applied too late, try `* { background-color: transparent !important; }` via native css (not css-in-js) and see how it behaves 🤷

**Comment by FabianLars** [comment]
Relevant for Windows: https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?tabs=dotnetcsharp#set-default-background-color-on-webview2-initialization-defaultbackgroundcolor-api should be part of the next stable sdk release whenever that'll be.

**Comment by txmazing** [comment]
> Relevant for Windows: https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?tabs=dotnetcsharp#set-default-background-color-on-webview2-initialization-defaultbackgroundcolor-api should be part of the next stable sdk release whenever that'll be.

v1.0.3296.44 with DefaultBackgroundColor was released on June 3, 2025.
https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?tabs=dotnetcsharp#10329644

**Comment by Aleel101** [comment]
> > Relevant for Windows: https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?tabs=dotnetcsharp#set-default-background-color-on-webview2-initialization-defaultbackgroundcolor-api should be part of the next stable sdk release whenever that'll be.
> 
> v1.0.3296.44 with DefaultBackgroundColor was released on June 3, 2025. https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?tabs=dotnetcsharp#10329644


Now i see that DefaultBackgroundColor was released on the 3rd, but Im lost on how to use this, I have two windows one for the main program and one for an overlay however when I show the overlay it flashes and to be honest I hate it. Can someone tell me how i can use this? Also a picture or code block of the fix would be great.

**Comment by FabianLars** [comment]
You can't. First of all the webview2 rust bindings have to be updated (tried but failed). Then the function must likely be used in tauri directly so it's set early enough.

Edit: nvm got it (the first part). dotnet just trolled me.

**Comment by Aleel101** [comment]
> You can't. First of all the webview2 rust bindings have to be updated (tried but failed). Then the function must likely be used in tauri directly so it's set early enough.
> 
> Edit: nvm got it (the first part). dotnet just trolled me.

wym you got it? you got it to work? if so do tell

**Comment by FabianLars** [comment]
"the first part" == updating the bindings crate. https://github.com/wravery/webview2-rs/pull/38

**Comment by FabianLars** [comment]
PR for the last tiny Windows white flash: https://github.com/tauri-apps/tao/pull/1125
Docs issue to write down my comments above in a guide: https://github.com/tauri-apps/tauri-docs/issues/3422

Since you can now configure the background color on all platforms and most of the flashes are gone (or can be removed via configs, which is what the comments are about). I'll close this issue. Other kinds of white flashes should get new gh issues (if there aren't any open ones).

**Comment by cmdmasquerade** [comment]
> PR for the last tiny Windows white flash: [tauri-apps/tao#1125](https://github.com/tauri-apps/tao/pull/1125) Docs issue to write down my comments above in a guide: [tauri-apps/tauri-docs#3422](https://github.com/tauri-apps/tauri-docs/issues/3422)
> 
> Since you can now configure the background color on all platforms and most of the flashes are gone (or can be removed via configs, which is what the comments are about). I'll close this issue. Other kinds of white flashes should get new gh issues (if there aren't any open ones).

When will this be merged?

**Comment by FabianLars** [comment]
That's arguably a shitty approach so perhaps never. I need someone who knows stuff about that part of windows to discuss it i guess.

Would be cool if some of you could test it though (NOT IN PRODUCTION!) by adding this to cargo.toml
```toml
[patch.crates-io]
tao = { git = "https://github.com/tauri-apps/tao", branch = "fix-windows-white-flash" }
```
will only work if you're on a reasonable recent tauri version.

**Comment by FTCHD** [comment]
@FabianLars 

is this a general improvement for tauri on windows, or is it only useful if we change the bg color per this issue?

**Comment by FabianLars** [comment]
should be a general improvement (on Windows) if you do not have a white background color in your app (frontend or tauri config) including transparent windows.

**Comment by FTCHD** [comment]
Been running it in prod for Rclone UI since your comment, no issues whatsoever

Could it be merged into **`dev`**?

**Comment by FabianLars** [comment]
Appreciate the feedback! I can use that as an argument when i talk with those that know more about tao than me :D
